circuitpython/bm_serial.py

```diff
@@ -0,0 +1,58 @@
+import board
+import busio
+
+
+class BristlemouthSerial:
+    def __init__(self, uart=None) -> None:
+        if uart is None:
+            self.uart = busio.UART(board.TX, board.RX, baudrate=115200)
+        else:
+            self.uart = uart
+
+    def spotter_tx(self, data) -> int | None:
+        packet = self.build_spotter_tx_packet(data)
+        return self.uart.write(packet)
+
+    def build_spotter_tx_packet(self, data: bytearray) -> bytes:
+        packet = (
+            bytearray.fromhex("020000001acccaf0eeeeffc001011500")
```

Comment:
is this the IP header?

Reply:
Great question. Thanks for asking.

The first 4 bytes are the `bm_serial_packet_t`. The initial `0x02` signifies a Bristlemouth publish packet type. The flags are zero, and 3rd and 4th bytes are the crc, which starts as zero, but gets calculated and overwritten on lines 22-24 below.

The next 12 bytes are the `bm_serial_pub_header_t`. I started with a silly node ID of `coffeeeef0cacc1a` (coffee focaccia), but I later realized that the node ID is ignored downstream, so these bytes can be anything. The cobs encoding and decoding is more efficient if there are no `0x00` bytes. After the node ID come the publish type and version, which should both be `0x01` for a spotter_tx. The last 2 bytes are a `uint16_t` topic length — "spotter/transmit-data" is 21 bytes long or 0x15 in hex. Both the node ID and the topic length are encoded little endian.

After the "spotter/transmit-data" topic there's one more `0x01` byte before the payload that corresponds to the `bm_serial_network_data_header_t`. Right now the only good value for that byte is cellular with Iridium fallback, but in the future when it's fully supported we could add the cellular-only option.